package.json

```diff
@@ -134,6 +132,7 @@
     "yarn-deduplicate": "^3.0.0"
   },
   "resolutions": {
-    "jest-expo/jest": "^26.4.1"
+    "jest-expo/jest": "^26.4.1",
+    "jsonfile": "^6.0.0"
```

Comment:
> Why is this line awful / why is it hilarious that things would break without it?

- It was a really tiny, obscure bit of code (node_modules/@expo/config/node_modules/@expo/configure-splash-screen/node_modules/fs-extra/lib/json/output-json.js) making Jest grind to a halt
- The code thinks it's fine to look inside a `utils.js` file in `output-json`, which is not part of its [documented public API](https://github.com/jprichardson/node-jsonfile)
- As you mention, "One thing that's unfortunate about it is that it would hold back dependencies that (in the future) want to take a later version of this library." But I thought that if I removed the upper bound (as in `>=6.0.0`) then that would override a potential upper bound that a dependency actually wanted to keep. (Now I'm actually not sure about this, hmm.)
- It seems to be relying on a bug in `yarn`. It looks like the following statement from the [doc](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/) isn't true: "You will receive a warning if your resolution version or range is not compatible with the original version range."

Reply:
Coming back to this again, though, actually it seems to work just fine without the `resolutions` line? And I agree, that commit in `fs-extra` looks like we shouldn't be seeing this problem. Not sure what I'd been doing wrong before. 🤷 

So I'll leave out this `resolutions` hack. 🙂

Reply:
Got it. Sounds good!

> * It seems to be relying on a bug in `yarn`. It looks like the following statement from the [doc](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/) isn't true: "You will receive a warning if your resolution version or range is not compatible with the original version range."

Yeah, that sounds familiar -- I feel like we've (or I've) run into that on some previous occasion using resolutions.

Even the documented behavior wouldn't have made this not work, though; it'd have made it emit a warning.


> But I thought that if I removed the upper bound (as in `>=6.0.0`) then that would override a potential upper bound that a dependency actually wanted to keep. (Now I'm actually not sure about this, hmm.)

Hmm, yeah -- that seems quite possible too. I guess this would be a good thing to confirm if in the future we want to use a resolution like this, at least for non-dev-only code.




> * The code thinks it's fine to look inside a `utils.js` file in `output-json`, which is not part of its [documented public API](https://github.com/jprichardson/node-jsonfile)

I see.

It looks like these two libraries are maintained together. So perhaps one way to look at this is that it's a symptom of the NPM style where everything is split into a thousand separate packages: what the author here really wanted was to have a small internal helper shared between these two bits of code, and if these were part of the same library that's exactly what they would have done, but they're not and they did it anyway. The proper thing to do given that they're separate packages would have been to think through that `stringify` helper's interface enough to document it as part of the API, but it gets tiring to do that all the time.